experimental/singular_standard_basis.jl

```diff
@@ -0,0 +1,73 @@
+mutable struct SingularStandardBasis{
```

Comment:
I actually don't think you gain anything from having all that type-parameters - but you loose readability. (Maybe @thofma  has a different opinion?) The elem types might have a performance benefit, but unless the others are used for dispatch, why?